src/multiplex/client.rs

```diff
@@ -117,18 +191,19 @@ impl<NT, Request> tower::load::Load for Maker<NT, Request> {
 /// request-at-a-time protocol transport. In particular, it wraps a transport that implements
 /// `Sink<SinkItem = Request>` and `Stream<Item = Response>` with the necessary bookkeeping to
 /// adhere to Tower's convenient `fn(Request) -> Future<Response>` API.
-pub struct Client<T, E, Request>
+pub struct Client<T, P, E, Request>
```

Comment:
I am considering putting P at the end and doing `P = VecDequePendingStore<T, Request>`, Not sure if we should go this way or not though.